contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,491 @@
+Additionally a set of label selectors will be added to each individual pod to ensure that the
+cluster routes the specific service traffic to the specific pod.
+The service type &#39;LoadBalancer&#39; will ensure that the data traffic can be routed across clusters.
+The network dns names for each pod will remain very similar to the network names used by the in-cluster statefulsets.
+
+As an example (using the same yamls and names as in the earlier par of this doc):
+
+For a federation with 3 clusters all in the same DNS zone _example.com_ and the statefulset
+name _store_ the dns names for instances might look like:
+```
+store-0.etcd.mynamespace.myfederation.svc.federation.example.com
+store-1.etcd.mynamespace.myfederation.svc.federation.example.com
+store-2.etcd.mynamespace.myfederation.svc.federation.example.com
+store-3.etcd.mynamespace.myfederation.svc.federation.example.com
+store-4.etcd.mynamespace.myfederation.svc.federation.example.com
+store-5.etcd.mynamespace.myfederation.svc.federation.example.com
+store-6.etcd.mynamespace.myfederation.svc.federation.example.com
+store-7.etcd.mynamespace.myfederation.svc.federation.example.com
+store-8.etcd.mynamespace.myfederation.svc.federation.example.com
+store-9.etcd.mynamespace.myfederation.svc.federation.example.com
+store-10.etcd.mynamespace.myfederation.svc.federation.example.com
+store-11.etcd.mynamespace.myfederation.svc.federation.example.com
+```
+This will be achieved by federated controller creating and monitoring individual services in
+specific clusters against each pod, with the same name as that pod&#39;s
+ordinal identity. It will further need to be ensured that the pods be assigned the same hostname
+as the pod name (ordinal identity) and service name to keep it consistent with the in-cluster
+statefulsets topology.
+
+The details about creation and handling of the LB services and updating the dns records will
+be similar to that elaborated in the previous design.
+
+This alternative has following drawbacks:
+ - As the control over the stateful pods is with the federation statefulset controller, the
+ monitoring of the stateful pods will not happen, if for some reason a cluster (or all clusters)
+ are disconnected from the federation control plane.
+ - Does not solve both the use cases listed at the beginning of this doc.
+
+Its thus proposed to use the primary design alternative listed earlier in this doc.
+
+# Federated statefulset updates
+
+The in cluster statefulsets update proposal is under implementation as per [this proposal](https://github.com/kubernetes/community/pull/503)
+The suggestion is to handle it as a separate design proposal, once the same is implemented and stabilised.
+
+
+# Limitations (to be discussed)
+
+_1 – In case of a particular replica or the cluster containing particular replicas die, what should be the behavior._
+
+It can be resurrected with the same identity in the original cluster only
```

Comment:
What is the implication of not being able to resurrect an identity?  i.e. if a cluster goes away and doesn't come back?